### DISCUSSION TITLE
IM N00B, trying the Weinhaus/IM/Cygwin route to image manip., Cygwin w/IM

### DISCUSSION THREAD

**Discussion by doojinsi** [ImageMagick/ImageMagick#2085] [Help]
Hope this is the right place, if not please shoo me away to the correct door.
I've been studying snibgo's excellent guide on working with the Weinhaus IM utils using IM and Cygwin under Win (10).   Not successfully up and running yet, couple of issues if anyone is working in this environment and has some wisdom.
1.  Got Win IM operational but the didn't get the convert.exe part.  I only have one convert.exe on my system -- it's the Win version.  Was I supposed to install the Cygwin version of IM?
2.  Got Cygwin operational (installed for all users) but went back to install a couple of packages like gnuplot, etc. using the installer.  The packages ARE NOT available in any bash shell on any folder in my system except if I run the cygwin.bat from C:\cygwin64 path.  Why would this be? Permissions?  Looks like the package doesn't exist, error is --
Command 'gnuplot' not found, but can be installed with:

sudo apt install gnuplot-nox  # version 5.2.8+dfsg1-2, or...
(and a minor question here is why does Cygwin tell you to use apt to install if it really wants you to use its' own installer?)
I know this is more Cygwin-related than IM, but this project is a several eggs in one basket.  Any ideas anyone.  Coffee?

**Comment by snibgo** [comment]
1. I don't understand. You "didn't get the convert.exe part", but you have "one convert.exe on my system"? Either you have convert.exe, or you don't.

As you have Windows, I suggest you install a pre-built v7 Windows binary from https://www.imagemagick.org/script/download.php#windows . You can tick the box that says "install legacy components (convert.exe etc)". I think Fred's scripts assume you have convert.exe.


2. What is in your system path? Mine looks like this:

PATH=C:\cygwin64\bin;C:\cygwin64\usr\local\bin;F:\pictures\bash; ...

You should probably ensure your IM installation directory is on your system path. (Personally, I normally don't, but I am weird.)


I suggest NOT installing IM from the Cygwin packages. Instead, use the link https://www.imagemagick.org/script/download.php#windows

Other packages, eg bc, sed, awk, gnuplot, etc can be installed using the Cygwin installer.

**Comment by doojinsi** [comment]
Mr. Gibson, thank you for the quick reply!

1.  The convert.exe that appears on my system is the Windows utility for disk manipulation.   When I enter convert at a cmd prompt I get the following message
Must specify a file system
and if I enter    where convert.exe    the result is 
C:\Windows\System32\convert.exe

2.  cygwin was NOT in my system path (curious, thought the installer would have shoved it in there), but IM was.  Taking your advice into consideration, I will proceed.

Many thanks,
Glen

p.s.  accept donations for your work/ideas/contributions?  if so, email me directly

**Comment by snibgo** [comment]
1. Put the IM installation directory at the front of your system path, or at least in front of the Windows directories. Then running "convert" will run the IM program, not the Microsoft program.

2. The Cygwin directories in your system path are needed so you can run gnuplot, sed, bc etc.

I don't accept donations.
